pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/ConcurrentMapPartitionUpsertMetadataManager.java

```diff
@@ -55,12 +64,17 @@ public class ConcurrentMapPartitionUpsertMetadataManager extends BasePartitionUp
 
   // Reused for reading previous record during partial upsert
   private final GenericRow _reuse = new GenericRow();
+  private long _lastExpiredTimeMS;
```

Comment:
javadoc? what does this variable mean?

Reply:
this is the watermark value. that keep track of the last time we cleaned up primary keys.

Reply:
updated this value to _largestSeenComparisonValueMs and added more clear definition.